packages/core/src/components/overlay/overlay.tsx

```diff
@@ -159,6 +159,19 @@ export class Overlay extends React.Component<IOverlayProps, IOverlayState> {
 
         const { children, className, inline, isOpen, transitionDuration, transitionName } = this.props;
 
+        // add a special class to each child that will automatically set the appropriate
+        // CSS position mode under the hood. we add this class *after* the existing classes
+        // so that our class will override the position style of other classes that might
```

Comment:
this is not how CSS works, which you call out later... it's entirely about specificity. if the user sets `position` in a selector with three classes it'll win out over ours every time.

i think you could safely remove all but the first sentence from this comment.